app/routes.py

```diff
@@ -1,2 +1,208 @@
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#create a task with null completed at
+@task_bp.route("", methods = ["POST"], strict_slashes = False)
+def create_task():
+    try:
+        request_body = request.get_json()
+        new_task = Task.from_json_to_task(request_body)
+        db.session.add(new_task) # "adds model to the db"
+        db.session.commit() # commits the action above
+        return new_task.to_json_response(), 201
+    except KeyError:
+        return {"details": "Invalid data"}, 400
+
+# Retrieve all /tasks  asc, desc, or id asc
+@task_bp.route("", methods = ["GET"], strict_slashes = False)
+def retrieve_tasks_data():
+    tasks = Task.query.all() 
+    sort_by = request.args.get("sort") # query parameters 
+    tasks_response = []
+
+    if tasks != None: 
+        if sort_by == "asc":  
+            # this is a list (queried by title) in asc order
+            tasks = Task.query.order_by(Task.title.asc()).all() 
+
+        elif sort_by == "desc":
+            # this is a list (queried by title) in desc order
+            tasks = Task.query.order_by(Task.title.desc()).all() 
+        
+        elif sort_by == "id":
+            # list, queried by id in asc order:
+            tasks = Task.query.order_by(Task.id.asc()).all()  
+        
+        tasks_response = [task.task_to_json_response() \
+                for task in tasks]
+    return jsonify(tasks_response), 200
+
+# Retrieve one /task/1     
+@task_bp.route("/<task_id>", methods=["GET"], strict_slashes = False)
+def retrieve_single_task(task_id):
+    task = Task.query.get(task_id)
+    if task != None:
+        # return task.to_json_response(), 200
+        response = task.to_json_response()
+        if task.goal_id:
+            response['task']['goal_id'] = task.goal_id
+        return response, 200
+    return make_response('', 404)
+
+#Update a task
+@task_bp.route("/<task_id>", methods=["PUT"], strict_slashes = False)  
+def update_task(task_id):
+    task = Task.query.get(task_id)
+    if task: # if successful quering task with given task_id
+        form_data = request.get_json()
+        if "title" in form_data:
+            task.title = form_data["title"]
+        if "description" in form_data:
+            task.description = form_data["description"]
+        if "completed_at" in form_data:
+            task.completed_at = form_data["completed_at"]
+        db.session.commit() # commiting changes to db
+        return task.to_json_response(), 200
+    return make_response(""), 404
+
+# Delete a task
+@task_bp.route("/<task_id>", methods=["DELETE"], strict_slashes = False)
+def delete_task(task_id):  
+    task = Task.query.get(task_id) 
+    if task != None:
+        db.session.delete(task)
+        db.session.commit()
+        details_str = f"Task {task_id} \"{task.title}\" successfully deleted"
+        return jsonify(details = details_str), 200
+    return make_response(""), 404
```

Comment:
This line looks a little different than the previous return of a 404. I'm not sure if it matters, but you could consider making a helper function to return a 404 to ensure that the ways you return a 404 are consistent.